tests/channels-search.bats

```diff
@@ -0,0 +1,165 @@
+}
+teardown() {
+  project_teardown;
+  common_test_teardown;
+}
+
+setup_file() {
+  export FLOX_FEATURES_CHANNELS=rust;
+}
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox search' can be called at all" {
+  run "$FLOX_CLI" search hello;
+  assert_success;
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox search' error with no search term" {
+  run "$FLOX_CLI" search;
+  assert_failure;
+}
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox search' helpful error with unquoted redirect: hello@>1 -> hello@" {
+  run "$FLOX_CLI" search hello@;
+  assert_failure;
+  assert_output --partial "try quoting";
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox search' expected number of results" {
+  run "$FLOX_CLI" search hello;
+  n_lines="${#lines[@]}";
+  assert_equal "$n_lines" "4"
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox search' semver search: hello@2.10" {
+  run "$FLOX_CLI" search hello@2.10;
+  assert_output --partial "hello.2_10";
+  n_lines="${#lines[@]}";
+  assert_equal "$n_lines" "1"
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox search' semver search: 'hello@>=1'" {
+  run "$FLOX_CLI" search 'hello@>=1';
+  assert_output --partial "hello.latest";
+  assert_output --partial "hello.2_12_1";
+  assert_output --partial "hello.2_12";
+  assert_output --partial "hello.2_10";
+  n_lines="${#lines[@]}";
+  assert_equal "$n_lines" "4"
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox search' semver search: hello@2.x" {
+  run "$FLOX_CLI" search hello@2.x;
+  assert_output --partial "hello.latest";
+  assert_output --partial "hello.2_12_1";
+  assert_output --partial "hello.2_12";
+  assert_output --partial "hello.2_10";
+  n_lines="${#lines[@]}";
+  assert_equal "$n_lines" "4"
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox search' semver search: hello@=2.10" {
+  run "$FLOX_CLI" search hello@=2.10;
+  assert_output --partial "hello.2_10";
+  n_lines="${#lines[@]}";
+  assert_equal "$n_lines" "1"
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox search' semver search: hello@v2" {
+  run "$FLOX_CLI" search hello@v2;
+  assert_output --partial "hello.2_12_1";
+  assert_output --partial "hello.2_12";
+  assert_output --partial "hello.2_10";
+  assert_output --partial "hello.latest";
+  n_lines="${#lines[@]}";
+  assert_equal "$n_lines" "4"
+}
+
+
+# ---------------------------------------------------------------------------- #
+
+@test "'flox search' semver search: 'hello@>1 <3'" {
+  run "$FLOX_CLI" search 'hello@>1 <3';
+  assert_output --partial "hello.2_12_1";
```

Comment:
I dont see us locking the catalog version that's being searched.
Assuming that we dont remove entries (which we plan to eventually), all tests with counting result lines are not fully controlled

Reply:
Fixed